sys/arvo.hoon

```diff
@@ -502,101 +515,232 @@
     $(ova (weld p.nyx ova), mor (weld q.nyx t.mor))
   --
 --
-=+  pit=`vase`!>(.)                                     ::
-=+  bud=pit                                             ::  becomes tang
-::  =+  vil=(viol p.bud)                                ::  cached reflexives
-=+  vil=(viol p.bud)                                    ::  cached reflexives
-=|  $:  lac/?                                           ::  laconic bit
-        eny/@                                           ::  entropy
+=<  ::  Arvo larval stage
+    ::
+    ::    The true Arvo kernel knows who it is. It should not *maybe*
+    ::    have an identity, nor should it contain multitudes. This outer
+    ::    kernel exists to accumulate identity, entropy, and the
+    ::    standard library. Upon having done so, it upgrades itself into
+    ::    the true Arvo kernel. Subsequent upgrades will fall through
+    ::    the larval stage directly into the actual kernel.
+    ::
+    ::    For convenience, this larval stage also supports hoon compilation
+    ::    with +wish and vane installation with the %veer event.
+    ::
+    =/  pit=vase  !>(..is)
+    =|  $:  ::  who: our identity once we know it
+            ::  eny: entropy once we learn it
+            ::  bod: %zuse once we receive it
+            ::
+            who=(unit ship)
+            eny=(unit @)
+            bod=(unit vase)
+        ==
+    ::  larval Arvo structural interface
+    ::
+    |%
+    ++  come  ^come                                     ::  22
+    ++  load  ^load                                     ::  46
+    ::
+    ::  XX my preference would be for +peek to no-op here
+    ::  XX falling thru because %vega scrys for hoon-version
+    ::  XX revisit
+    ::
+    ++  peek  ^peek                                     ::  47
+    ::
+    ++  poke  |=  *                                     ::  10
+              ^-  [(list ovum) *]
+              =>  .(+< ((hard ,[now=@da ovo=ovum]) +<))
+              ^-  [(list ovum) *]
+              =.  +>.$
+                ?+  -.q.ovo
+                  ::  ignore unrecognized
+                  ::
+                  ~&  [%larval-ignore p.ovo -.q.ovo]
+                  +>.$
+                ::  install %zuse or vane
+                ::
+                    %veer
+                  ^+  +>.$
+                  ::  use the maximum comet if we don't know who we are yet
+                  ::
+                  =/  our  ?^(who u.who (dec (bex 128)))
```

Comment:
I feel like this should emit a debug message if we're assuming a temporary identity. Otherwise bugs related to this could be hard to figure out.

Reply:
Do we really need this feature? As part of locking this down, I wouldn't be surprised if we want to remove the ability for a larval Arvo to install vanes. That would also let us bake the `our` identity into a vane at install time, which feels right to me.

Reply:
The debug message is good idea. We need the feature for a `+solid` pill.